framework/base/Application.php

```diff
@@ -246,6 +246,18 @@ public function preInit(&$config)
             $this->setTimeZone('UTC');
         }
 
+        if (isset($config['container'])) {
```

Comment:
Another option can be introduction of the single setter named `setContainer()`, which will always accept array with `definitions` and `singletons` keys.